wlcreate/Start-Asian-Love
# Resolves 15. As a admin, I'd like to have a backend for our website

## Description

Related Issue: 
closes #37 

This PR merges to main our official transition to Sanity as our headless CMS. The `About`, `Support`, `Wins` and `WinsID` pages fetch from Sanity using `getStaticProps`, and `WinsID` additionally fetches its paths through `getStaticPaths`. Instead of nesting routes/pages for each `WinsID` and repeating code, we are dynamically generating each page/path with `[slug].js`. Something important to note since the previous PRs related to this issue is that since there are not many paths we have decided to set `isFallback` to false for now.

Furthermore, decisions to separate components were made. In `_app.js` we are now rendering the `<Header />` and `<Footer />` components there to avoid importing them in each file. For each time we are mapping over resources (ex: Profiles, Support Resources, or Wins Resources) we have decided to have the container of the resources be its own separate component from the individual (ex: Profile, Support Resource, or Wins Resource). To account for the growing file structure, within `styles` there are folders that specify if the files within are `components` or `page` therefore making it easier to navigate the stylesheets. 

Finally, since we have made this transition to Sanity, the `database` folder and profile photos in `public` have been removed. 

Future things to think about:
- if someone is viewing in `incognito` mode, `WinsID` does not show because of `preview: true`
- Error 404 page
- loading component
- potentially making `isFallback` true 

Resources:
- [sanity docs](https://www.sanity.io/docs/getting-started)
- [next-sanity package](https://github.com/sanity-io/next-sanity)
- view PR #43
- view PR #44 

## Acceptance Criteria


Acceptance Criteria: 

- Use a headless CMS such as [Sanity](https://www.sanity.io/docs/getting-started):
  -  Import Wins and Support resources onto our newly created database
  -  Decide how we want to structure our data (esp for support resources)
  -  Deploy it
  -  Figure out how to connect frontend code with our new backend
  -  Update the codebase as needed for getting and displaying our data

Resources:
- https://www.sanity.io/docs/getting-started
- https://www.youtube.com/watch?v=32RP-sG1njE
- https://www.youtube.com/watch?v=1WmNXEVia8I

## Self-check

Please check all that apply:




## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|  ✓  | :scroll: Docs              |

## Updates

### Before



### After



## Testing Steps / QA Criteria


- visit this branch `git checkout sanity-backend`
- get the latest changes `git pull origin sanity-backend`
- run the development `npm run dev`
- open browser to `localhost:3000`
- check if can view all About, Support, Wins, and WinsID pages
- when visiting a route that does not exist, seeing a 404 page
